relay-server/src/utils/error_boundary.rs

```diff
@@ -1,9 +1,42 @@
-use std::fmt::Debug;
+use std::fmt::{self, Debug};
 
 use failure::Fail;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use serde_json::Value;
 
+/// Structure used in the implementation of Clone for ErrorBoundary
+/// Couldn't find a nicer way to do it.
+/// For the case where ErrorBoundary contains an error (Err enum) I copy
+/// the representation of the Fail trace object inside a ClonedFail object
+/// during a clone, that means that I loose the original Fail object and only
+/// retain the text message.
+#[derive(Fail, Debug, Clone)]
+struct ClonedFail(String);
```

Comment:
This loses error information since it stringifies the error. It would be better to constrain the error type to `Clone`, which can be achieved with a utility trait.

Reply:
Yes, it does, I didn't know how to do it nicer, I spent a bit of time with @untitaker on that but couldn't find a clean way and I was not very concerned about loosing the original error.
Maybe we can pair-program on this and clean it up.